### PR TITLE
sat up server side and created API route

### PR DESCRIPTION
**Pull request by DwinaTech**

### CODE REVIEW

File: backend/app.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+var bodyParser = require('body-parser');
+
+var index = require('./routes/index');
+var users = require('./routes/users');
+var router = require('./routes/api');
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
```
Comment (kkarimi) [inline backend/app.js]: We do not need handlebars as of now

File: backend/app.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+var express = require('express');
+var path = require('path');
+var favicon = require('serve-favicon');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
```
Comment (kkarimi) [inline backend/app.js]: We do not need cookie parser.. yet

File: backend/app.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+var index = require('./routes/index');
+var users = require('./routes/users');
+var router = require('./routes/api');
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'hbs');
```
Comment (kkarimi) [inline backend/app.js]: ditto

File: backend/app.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+// uncomment after placing your favicon in /public
+//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
```
Comment (kkarimi) [inline backend/app.js]: ditto

File: backend/public/stylesheets/style.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+body {
```
Comment (kkarimi) [inline backend/public/stylesheets/style.css]: do not need any frontend from backend, all frontend will be handled by React

File: backend/views/error.hbs [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<h1>{{message}}</h1>
```
Comment (kkarimi) [inline backend/views/error.hbs]: do not need the `views` directory

File: backend/routes/users.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
```
Comment (kkarimi) [inline backend/routes/users.js]: only need `api` routes, not any other routes

File: backend/routes/index.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
+var router = express.Router();
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  res.render('index', { title: "DOS backend" });
```
Comment (kkarimi) [inline backend/routes/index.js]: we will only be sending `json` no need for any `renders`

File: backend/package.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "backend",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "start": "node ./bin/www",
+    "dev": "nodemon ./bin/www"
```
Comment (kkarimi) [inline backend/package.json]: nice!

File: backend/package.json [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "dependencies": {
+    "body-parser": "~1.18.2",
+    "cookie-parser": "~1.4.3",
+    "debug": "~2.6.9",
+    "express": "~4.15.5",
+    "hbs": "~4.0.1",
```
Comment (kkarimi) [inline backend/package.json]: please remove:

1.  service-favicon
1. cookie-parser
1. hbs

File: backend/bin/www [hunk 11]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+
+/**
+ * Module dependencies.
+ */
+
+var app = require('../app');
+var debug = require('debug')('backend:server');
```
Comment (kkarimi) [inline backend/bin/www]: I think you can use morgan style debug and then no need for `debug` here

File: backend/bin/www [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * Event listener for HTTP server "listening" event.
+ */
+
+function onListening() {
+  var addr = server.address();
+  var bind = typeof addr === 'string'
+    ? 'pipe ' + addr
+    : 'port ' + addr.port;
+  debug('Listening on ' + bind);
```
Comment (kkarimi) [inline backend/bin/www]: use morgan/logger here

Comment (DwinaTech) [reply backend/bin/www]: Done

File: backend/bin/www (hunk 11 above)
Comment (DwinaTech) [reply backend/bin/www]: Done

File: backend/package.json (hunk 10 above)
Comment (DwinaTech) [reply backend/package.json]: Done

File: backend/routes/index.js (hunk 8 above)
Comment (DwinaTech) [reply backend/routes/index.js]: Done

File: backend/public/stylesheets/style.css (hunk 5 above)
Comment (DwinaTech) [reply backend/public/stylesheets/style.css]: Done

File: backend/routes/users.js (hunk 7 above)
Comment (DwinaTech) [reply backend/routes/users.js]: Done

File: backend/app.js (hunk 4 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/app.js (hunk 3 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/app.js (hunk 2 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/app.js (hunk 1 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/views/error.hbs (hunk 6 above)
Comment (DwinaTech) [reply backend/views/error.hbs]: Done

File: backend/app.js [hunk 13]
```diff
@@ -0,0 +1,40 @@
+var express = require('express');
+var bodyParser = require('body-parser');
+var path = require('path')
+const morgan = require('morgan');
+var morganDebug = require('morgan-debug');
```
Comment (kkarimi) [inline backend/app.js]: @DwinaTech Morgan should be enough.. `morgan-debug` seems unnecessary

File: backend/app.js [hunk 14]
```diff
@@ -0,0 +1,40 @@
+var express = require('express');
```
Comment (kkarimi) [inline backend/app.js]: All `vars` here should be `const`

File: backend/app.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+var router = require('./routes/api');
+
+var app = express();
+
+// Show requestes with morgan
+app.use(morgan('combined'));
+
+//Morgan debug
+app.use(morganDebug('backend', 'combined'));
```
Comment (kkarimi) [inline backend/app.js]: `Morgan` should be enough

File: backend/bin/www [hunk 16]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+/**
+ * Module dependencies.
+ */
+
+var app = require('../app');
```
Comment (kkarimi) [inline backend/bin/www]: all vars should be `const`

File: backend/package.json [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+  "scripts": {
+    "start": "node ./bin/www",
+    "dev": "nodemon ./bin/www"
+  },
+  "dependencies": {
+    "body-parser": "~1.18.2",
+    "debug": "~2.6.9",
+    "express": "~4.15.5",
+    "morgan": "^1.9.0",
+    "morgan-debug": "^2.0.0",
```
Comment (kkarimi) [inline backend/package.json]: ditto

File: backend/routes/api.js [hunk 18]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
```
Comment (kkarimi) [inline backend/routes/api.js]: vars should be `const`

File: backend/app.js (hunk 13 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/app.js (hunk 14 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/app.js (hunk 15 above)
Comment (DwinaTech) [reply backend/app.js]: Done

File: backend/bin/www (hunk 16 above)
Comment (DwinaTech) [reply backend/bin/www]: Done

File: backend/package.json (hunk 17 above)
Comment (DwinaTech) [reply backend/package.json]: Done

File: backend/routes/api.js (hunk 18 above)
Comment (DwinaTech) [reply backend/routes/api.js]: Done

**Pull request merged**